include/libdnf/base/goal.hpp

```diff
@@ -54,53 +54,285 @@ class Goal {
     void add_rpm_distro_sync();
+
+    // TODO(jmracek) Add suport `to_repo_ids`
+    /// Add distrosync job of all installed packages to the goal. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param settings  A structure to override default goal settings. Values `from_repo_ids` is not used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_distupgrade_all(HyGoal goal)
+    // @replaces dnf:dnf/base.py:method:Base().distro_sync(self, pkg_spec=None)
     void add_rpm_distro_sync(const libdnf::GoalJobSettings & settings);
+
+    /// Add distrosync job to the goal. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param rpm_package      A package to be added to the distrosync job.
+    /// @param settings         A structure to override default goal settings. Only `clean_requirements_on_remove` and `best` values are used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_distupgrade(HyGoal goal, DnfPackage *new_pkg)
     void add_rpm_distro_sync(const libdnf::rpm::Package & rpm_package);
+
+    /// Add distrosync job to the goal. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param rpm_package      A package to be added to the distrosync job.
+    /// @param settings         A structure to override default goal settings. Only `strict`, `best`, and `clean_requirements_on_remove` values are used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_distupgrade(HyGoal goal, DnfPackage *new_pkg)
     void add_rpm_distro_sync(const libdnf::rpm::Package & rpm_package, const libdnf::GoalJobSettings & settings);
+
+    /// Add distrosync job to the goal. In default it uses `clean_requirements_on_remove` with `false` value.
+    ///
+    /// @param package_set      A package_set to be added to the distrosync job.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_distupgrade_selector(HyGoal goal, HySelector)
     void add_rpm_distro_sync(const libdnf::rpm::PackageSet & package_set);
+
+    /// Add distrosync job to the goal. In default it uses `clean_requirements_on_remove` with `false` value,
+    /// but it can be overridden by setting.
+    ///
+    /// @param package_set      A package_set to be added to the distrosync job.
+    /// @param settings         A structure to override default goal settings. Only `strict`, `best`, and `clean_requirements_on_remove` values are used.
+    // @replaces libdnf/hy-goal.h:function:hy_goal_distupgrade_selector(HyGoal goal, HySelector)
     void add_rpm_distro_sync(const libdnf::rpm::PackageSet & package_set, const libdnf::GoalJobSettings & settings);
 
+    // TODO(jmracek) Move transaction reports to Transaction class
+    /// Resolve all jobs and return a transaction object
+    ///
+    /// @param allow_erasing    A bool value required for solver
```

Comment:
_"A bool value required for solver"_ isn't really a proper description of what the argument is.

Reply:
I updated the description